components/console/single_command_tool.rst

```diff
@@ -6,68 +6,51 @@ Building a single Command Application
-it is possible to remove this need by extending the application::
-
-    namespace Acme\Tool;
-
-    use Symfony\Component\Console\Application;
-    use Symfony\Component\Console\Input\InputInterface;
-
-    class MyApplication extends Application
-    {
-        /**
-         * Gets the name of the command based on input.
-         *
-         * @param InputInterface $input The input interface
-         *
-         * @return string The command name
-         */
-        protected function getCommandName(InputInterface $input)
-        {
-            // This should return the name of your command.
-            return 'my_command';
-        }
-
-        /**
-         * Gets the default commands that should always be available.
-         *
-         * @return array An array of default Command instances
-         */
-        protected function getDefaultCommands()
-        {
-            // Keep the core default commands to have the HelpCommand
-            // which is used when using the --help option
-            $defaultCommands = parent::getDefaultCommands();
-
-            $defaultCommands[] = new MyCommand();
-
-            return $defaultCommands;
-        }
-
-        /**
-         * Overridden so that the application doesn't expect the command
-         * name to be the first argument.
-         */
-        public function getDefinition()
-        {
-            $inputDefinition = parent::getDefinition();
-            // clear out the normal first argument, which is the command name
-            $inputDefinition->setArguments();
-
-            return $inputDefinition;
-        }
-    }
-
-When calling your console script, the command ``MyCommand`` will then always
-be used, without having to pass its name.
-
-You can also simplify how you execute the application::
-
-    #!/usr/bin/env php
-    <?php
-    // command.php
-
-    use Acme\Tool\MyApplication;
-
-    $application = new MyApplication();
-    $application->run();
+it is possible to remove this need by declaring a single command application::
+
+  #!/usr/bin/env php
+  <?php
+  require __DIR__.'/vendor/autoload.php';
+
+  use Symfony\Component\Console\Application;
+  use Symfony\Component\Console\Input\InputArgument;
+  use Symfony\Component\Console\Input\InputInterface;
+  use Symfony\Component\Console\Input\InputOption;
+  use Symfony\Component\Console\Output\OutputInterface;
+
+  (new Application('echo', '1.0.0'))
+    ->register('echo')
+        ->addArgument('foo', InputArgument::OPTIONAL, 'The directory')
+        ->addOption('bar', null, InputOption::VALUE_REQUIRED)
+        ->setCode(function(InputInterface $input, OutputInterface $output) {
+            // output arguments and options
+        })
+    ->getApplication()
+    ->setDefaultCommand('echo', true) // Single command application
+    ->run();
+
+The method :method:`Symfony\\Component\\Console\\Application::setDefaultCommand`
+accepts a boolean as second parameter. If true, the command ``echo`` will then
```

Comment:
Should we elaborate a bit on this second argument? I guess that it may sound strange that you have to pass this option at all (one might expect from the name of the `setDefaultCommand()` method that this always is the case anyway).